api/src/pcapi/core/operations/api.py

```diff
@@ -42,3 +52,108 @@ def create_special_event_from_typeform(
+                    "exc": str(exc),
+                },
+            )
+
+
+@atomic()
+def update_form_title_from_typeform(event: models.SpecialEvent, form: typeform.TypeformForm) -> None:
+    if form.title != event.title:
+        event.title = form.title
+        db.session.add(event)
+        db.session.flush()
+
+
+@atomic()
+def update_form_questions_from_typeform(
+    event: models.SpecialEvent, form: typeform.TypeformForm
+) -> dict[str, models.SpecialEventQuestion]:
+    old_questions = models.SpecialEventQuestion.query.filter(models.SpecialEventQuestion.eventId == event.id).all()
+
+    questions = {q.externalId: q for q in old_questions}
+
+    for question in form.fields:
+        if question.field_id not in questions:
+            questions[question.field_id] = models.SpecialEventQuestion(
+                eventId=event.id, externalId=question.field_id, title=question.title
+            )
+            db.session.add(questions[question.field_id])
+            db.session.flush()
+        elif question.title != questions[question.field_id].title:
+            questions[question.field_id].title = question.title
+            db.session.add(questions[question.field_id])
+            db.session.flush()
+    return questions
+
+
+def download_responses_from_typeform(
+    event: models.SpecialEvent, questions: dict[str, models.SpecialEventQuestion]
+) -> None:
+
+    def get_last_date_for_event() -> datetime | None:
+        result = (
+            models.SpecialEventResponse.query.with_entities(models.SpecialEventResponse.dateSubmitted)
+            .filter(models.SpecialEventResponse.eventId == event.id)
+            .order_by(models.SpecialEventResponse.dateSubmitted.desc())
+            .limit(1)
+            .scalar()
+        )
+        return result
+
+    for response in typeform.get_responses_generator(get_last_date_for_event, event.externalId):
+        save_response(
+            event=event,
+            form=response,
+            questions=questions,
+        )
+
+
+@atomic()
+def save_response(
+    event: models.SpecialEvent, form: typeform.TypeformResponse, questions: dict[str, models.SpecialEventQuestion]
+) -> None:
+    try:
+        response = models.SpecialEventResponse(
+            eventId=event.id,
+            externalId=form.response_id,
+            dateSubmitted=form.date_submitted,
+            phoneNumber=form.phone_number,
+            email=form.email,
+        )
+        db.session.add(response)
+        db.session.flush()
+    except sa.exc.IntegrityError:
+        mark_transaction_as_invalid()
+        return
+
+    for answer in form.answers:
+        if answer.field_id not in questions:
+            continue
+        if not answer.text:
+            continue
+        db.session.add(
+            models.SpecialEventAnswer(
+                responseId=response.id,
+                questionId=questions[answer.field_id].id,
+                text=answer.text or "",
```

Comment:
On a `if not answer.text:` au-dessus, donc selon ma compréhension, on peut se limiter à : 
```suggestion
                text=answer.text,
```